service/controller/legacy/v29/ebs/ebs.go

```diff
@@ -128,6 +128,43 @@ func (e *EBS) DetachVolume(ctx context.Context, volumeID string, attachment Volu
 		if err != nil {
 			return microerror.Mask(err)
 		}
+	}
+
+	{
+		i := &ec2.DescribeVolumesInput{
+			VolumeIds: []*string{
+				aws.String(volumeID),
+			},
+		}
+
+		v, err := e.client.DescribeVolumes(i)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		if len(v.Volumes) == 0 {
```

Comment:
Why did you change the conditions? There are maybe less or more than 1 volume and we want to return an error in both cases I think.

Reply:
Because we spoke during our call and you said clearly that if we do the lookup with the volumeID then the only possible result is one volume. Not more than one. So i removed that condition.

Reply:
We said there should only be one volume and if there is none or more than one we should return an error. I made comments for that. The conditions have been correct as you implemented them, only the error handling needed improvements.